File: tests/fixtures/language_pack/src/kotlin/Widget.kt
package languagepack.kotlin

import kotlin.collections.List

typealias KotlinName = String

enum class KotlinMode {
    Compact
}

class KotlinWidget(
    val name: KotlinName,
) {
    val KotlinLimit = 4

    fun render(
        prefix: String,
    ): String {
        val ignored = "class KotlinStringFake"
        return prefix + name
    }
}

object KotlinWidgetFactory {
    fun buildKotlinWidget(
        name: KotlinName,
    ): KotlinWidget {
        return KotlinWidget(name)
    }
}

File: tests/fixtures/language_pack/src/scala/Widget.scala
package languagepack.scala

import scala.collection.mutable.ListBuffer

trait ScalaRenderable {
  def render(): String
}

enum ScalaMode {
  case Compact
}

class ScalaWidget(name: String) extends ScalaRenderable {
  val ScalaLimit = 4
  var scalaState = name

  def render(
      prefix: String
  ): String = {
    val ignored = "class ScalaStringFake"
    prefix + name
  }
}

object ScalaWidgetFactory {
  type ScalaName = String

  def buildScalaWidget(
      name: String
  ): ScalaWidget = ScalaWidget(name)
}
